dali/pipeline/operator/arg_helper.h

```diff
@@ -116,7 +116,7 @@ class ArgValue {
       DALI_ENFORCE(is_uniform(view_.shape) && expected_shape == view_.shape[0],
         make_string("Expected uniform shape for argument \"", arg_name_,
                     "\" but got shape ", view_.shape));
-    } else {
+    } else if (view_.empty()) {
```

Comment:
1. It assumes that non-tensor data is constant (and we may change that at some point)
2. It validates the hidden (and dangerous) assumption that Acquire is no-op - I think that hoisting this logic outside of Acquire violates encapsulation of ArgValue.
3. It makes clear that the way we obtain non-tensor arguments is inefficient, as it results in reallocation of the internal vector - use TryGetArgument / TryGetRepeatedArgument which will resize and populate an existing vector.